File: python/1827.py
while True:
    try:
        n = int(input())
        i, j = 0, 0
        o = ''
        k = n//3

        while i < n:
            while j < n:
                if i == n//2 and j == n//2:
                    o += '4'
                elif j >= k and i >= k and j <= (n-1)-k and i <= (n-1)-k:
                    o += '1'
                elif j == i:
                    o += '2'
                elif j == n-i-1:
                    o += '3'
                else:
                    o += '0'
                j += 1
            i += 1
            j = 0
            o += '\n'
        print(o)
    except EOFError:
        break

File: dart/1827.dart
import 'dart:io';

void main() {
  while (true) {
    var input = stdin.readLineSync();

    if (input == null) {
      break;
    }

    int n = int.parse(input);
    String output = "";
    int k = n~/3;

    for (int i = 0; i < n; i++) {
      for (int j = 0; j < n; j++) {
        if (i == n~/2 && j == n~/2) {
          output += "4";
        } else if (j >= k && i >= k && j <= (n-1)-k && i <= (n-1)-k) {
          output += "1";
        } else if (i == j) {
          output += "2";
        } else if (j == n-i-1) {
          output += "3";
        } else {
          output += "0";
        }
      }
      output += "\n";
    }

    print(output);
  }
}
